org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/CompilationUnit.java

```diff
@@ -610,10 +650,12 @@ static IBinding resolveBinding(ASTNode node) {
 	if (node instanceof ConstructorInvocation constructor) {
 		return constructor.resolveConstructorBinding();
 	}
+	if (node instanceof org.eclipse.jdt.core.dom.Annotation annotation) {
+		return annotation.resolveTypeBinding();
+	}
```

Comment:
Can you maybe just submit this in a dedicated PR. It shouldn't cause any issue and will fix a couple bugs.

Reply:
Here it is #112